src/entity/EntityFactory.php

```diff
@@ -81,6 +82,14 @@ public function __construct(){
 		//define legacy save IDs first - use them for saving for maximum compatibility with Minecraft PC
 		//TODO: index them by version to allow proper multi-save compatibility
 
+		$this->register(AreaEffectCloud::class, function(World $world, CompoundTag $nbt) : AreaEffectCloud{
+			$potionType = PotionTypeIdMap::getInstance()->fromId($nbt->getShort("PotionId", PotionTypeIds::WATER));
+			if($potionType === null){
+				throw new SavedDataLoadingException("No such potion type");
+			}
+			return new AreaEffectCloud(Helper::parseLocation($nbt, $world), $potionType, $nbt);
+		}, ['AreaEffectCloud', 'minecraft:area_effect_cloud'], LegacyIds::AREA_EFFECT_CLOUD);
```

Comment:
Are you sure these are correct? Don't make any assumptions - sometimes the legacy IDs in particular have seriously weird names.

Reply:
Yeah, vanilla nbt tree:
![image](https://user-images.githubusercontent.com/57236932/190538649-aad8664f-790d-4e1a-8801-9b8de5b08661.png)